From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Geert Uytterhoeven <geert+renesas@glider.be>
Date: Tue, 9 Jun 2026 17:08:34 +0200
Subject: [PATCH 126/220] pinctrl: tegra: PINCTRL_TEGRA238 should depend on
 ARCH_TEGRA

The NVIDIA Tegra238 MAIN and AON pin controllers are only present on
NVIDIA Tegra238 SoCs.  Hence add a dependency on ARCH_TEGRA, to prevent
asking the user about this driver when configuring a kernel without
NVIDIA Tegra SoC support.

Fixes: 25cac7292d49f4fc ("pinctrl: tegra: Add Tegra238 pinmux driver")
Signed-off-by: Geert Uytterhoeven <geert+renesas@glider.be>
Reviewed-by: Krzysztof Kozlowski <krzysztof.kozlowski@oss.qualcomm.com>
Signed-off-by: Linus Walleij <linusw@kernel.org>
(cherry picked from commit 5f899621d34e2fd79229622805a659b13b63c98f)

diff --git a/drivers/pinctrl/tegra/Kconfig b/drivers/pinctrl/tegra/Kconfig
index 54263b4554c0..4fa7295ae449 100644
--- a/drivers/pinctrl/tegra/Kconfig
+++ b/drivers/pinctrl/tegra/Kconfig
@@ -38,6 +38,7 @@ config PINCTRL_TEGRA234
 
 config PINCTRL_TEGRA238
 	tristate "NVIDIA Tegra238 pinctrl driver"
+	depends on ARCH_TEGRA || COMPILE_TEST
 	default m if ARCH_TEGRA_238_SOC
 	select PINCTRL_TEGRA
 	help
-- 
2.43.0

